Tools/check_submodules.sh

```diff
@@ -61,34 +59,26 @@ then
 	[ -n "$GIT_SUBMODULES_ARE_EVIL" ] && {
 		# GIT_SUBMODULES_ARE_EVIL is set, meaning user doesn't want submodules updated
 		echo "GIT_SUBMODULES_ARE_EVIL is defined - Skipping submodules $1 update."
-	  exit 0
+		exit 0
 	}
 
 	git submodule update --recursive $1
 
 else
 
 	[ -n "$GIT_SUBMODULES_ARE_EVIL" ] && {
-	  # GIT_SUBMODULES_ARE_EVIL is set, meaning user doesn't want submodules updated
-	  echo "GIT_SUBMODULES_ARE_EVIL is defined - Skipping All submodule checking!"
-	  exit 0
+		# GIT_SUBMODULES_ARE_EVIL is set, meaning user doesn't want submodules updated
+		echo "GIT_SUBMODULES_ARE_EVIL is defined - Skipping All submodule checking!"
+		exit 0
 	}
 
-	check_git_submodule NuttX
-	check_git_submodule Tools/gencpp
-	check_git_submodule Tools/genmsg
-	check_git_submodule Tools/jMAVSim
-	check_git_submodule Tools/sitl_gazebo
-	check_git_submodule cmake/cmake_hexagon
-	check_git_submodule mavlink/include/mavlink/v1.0
-	check_git_submodule mavlink/include/mavlink/v2.0
-	check_git_submodule src/lib/DriverFramework
-	check_git_submodule src/lib/DriverFramework/cmake/cmake_hexagon
-	check_git_submodule src/lib/DriverFramework/dspal
-	check_git_submodule src/lib/ecl
-	check_git_submodule src/lib/matrix
-	check_git_submodule src/modules/uavcan/libuavcan
-	check_git_submodule unittests/googletest
-	check_git_submodule src/drivers/gps/devices
+	declare -a submodules=$(git submodule status --recursive | awk '{ print $2 }')
+	mapfile -t arr <<< "$submodules"
```

Comment:
from one link i found that:
"Since bash4, this can be done even more efficiently with the mapfile builtin"